src/helpers/database.py

```diff
@@ -173,11 +173,20 @@ def write_prices_new(self, prices: list[tuple[str, int, float, str]]) -> None:
         )
         for token_address, time, price, source in prices:
             try:
+                date_time = datetime.fromtimestamp(time, tz=timezone.utc)
+                check_existence_query = f"SELECT * FROM prices WHERE token_address = '\\{token_address[1:]}' and time = '{date_time}' and source = '{source}'"
+                result = self.execute_query(check_existence_query, {}).fetchone()
```

Comment:
I would probably move those two lines into its own function and conform to the format suggested by `execute_query`. I.e. something like
```python
def get_prices(self, token_address, time, source):
    query = "SELECT * FROM prices WHERE token_address = :token_address AND time = :time AND source = :source"
    result = self.execute_query(
        query,
        {
            "token_address": bytes.fromhex(token_address[2:]),
            "time": datetime.fromtimestamp(time, tz=timezone.utc),
            "source": source,
        }
    )
    return result.fetchall()
```
(not tested)

Reply:
Addressed this